### DISCUSSION TITLE
Placing points on Path3D is so tedious.

### DISCUSSION THREAD

**Discussion by Yusbro** [godotengine/godot-proposals#7708] [3D]
it is so tedious!, it doesn't place points *Over* a collider or something. nope, places points *Through* the collider. 
it may not sound like a big problem. but it gets so annoying when you're trying to place a bunch of points inside a multi-storey building a bunch of enemies that can move between different floors. but you can't really do that with Path3D, because the points completely pass through any collider. so if you wanna do that, you have to manually drag it around. which is so hard when you're working with a bit tricky level. so yeah, I think it'll be nice if Path3D has a tiny "Raycast" option. which kinda places points over the level geometry. and not place it inside them or somewhere else. 

[Trying to place a bunch of Points over the level]
![image](https://github.com/godotengine/godot-proposals/assets/100201070/50f48357-e504-4027-987f-0d7f74baffdb)

[Oopsie!. the points were place beyond the level geometry]
![image](https://github.com/godotengine/godot-proposals/assets/100201070/d15117f5-891d-4ed2-84e0-fbf3f0dcd095)

**Comment by Calinou** [comment]
Usability improvements for Path3D gizmo are being added to https://github.com/godotengine/godot/pull/80802.

Placing based on collision or geometry is something that will have to be done in a separate PR though. It may be better to look into adding general vertex snapping functionality first.

In the meantime, what you can do is create a in-game helper to create Curve3D resources saved to `.tres` files based on your in-game position when you press a certain key.
